pkg/analyzer/detection/detection.go

```diff
@@ -0,0 +1,139 @@
+package detection
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/stackrox/scanner/database"
+	"github.com/stackrox/scanner/ext/featurefmt"
+	"github.com/stackrox/scanner/ext/featurens"
+	"github.com/stackrox/scanner/pkg/analyzer"
+	"github.com/stackrox/scanner/pkg/component"
+	features2 "github.com/stackrox/scanner/pkg/features"
+	"github.com/stackrox/scanner/pkg/rhelv2/rpm"
+	"github.com/stackrox/scanner/pkg/wellknownnamespaces"
+)
+
+// LogLayerName is the name of the log field holding the detection target.
+const LogLayerName = "layer"
+
+// DetectComponents detects the namespace and extracts the components present in
+// the files of a filesystem or image layer. For layers, the parent layer should
+// be specified. For filesystems, which don't have the concept of intermediate
+// layers, or the root layer, use `nil`. Notice that language components are not
+// extracted by DetectComponents, but if provided they are annotated with
+// certified RHEL dependencies, and returned.
+func DetectComponents(name string, files analyzer.Files, parent *database.Layer, languageComponents []*component.Component, uncertifiedRHEL bool) (*database.Namespace, []database.FeatureVersion, *database.RHELv2Components, []*component.Component, error) {
```

Comment:
I'm assuming this is simply copy/pasted (aside from the function's rename)?

Reply:
Moved around the `name` to be the first parameter to match `DetectNamespace`. Other than that, yes, this is a copy'n'paste.